src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -243,6 +224,63 @@
      */
     InventoryTransactionResult set(ItemStack stack);
 
+    /**
+     * Gets and remove the stack at the supplied index in this Inventory.
```

Comment:
These methods need to be clear on when an `Optional` is empty or not.

Because, we can simplify this to `ItemStack.EMPTY` if we're returning it for a valid slot index, and `Optional.empty()` when the `SlotIndex` is invalid. OR, simplify these methods to always return an `ItemStack` or throw an exception if the slot index is invalid. My two cents on the matter.
The optionals are somewhat annoying to work with, and frankly, it's easier to say "Hey, you're working on an empty itemstack," versus "Hey, you might be working with an empty itemstack anyways, but you have to check the optional first!"

Reply:
I changed it to `ItemStack` instead of `Optional` for the ones without SlotIndex.
With SlotIndex still returns Optional because the Slot might not even exist.
`inv.peek(SlotIndex.of(1))` is the same as `inv.getSlot(SlotIndex.of(1)).map(Slot::peek)`